system/TestBox.cfc

```diff
@@ -50,19 +52,60 @@ component accessors="true"{
 		variables.reporter = arguments.reporter;
 		// options
 		variables.options = arguments.options;
+		// Empty bundles to start
+		variables.bundles = [];
 
+		// inflate labels
+		inflateLabels( arguments.labels );
+		// add bundles
+		addBundles( arguments.bundles );
+		// Add directory given (if any)
+		addDirectory( arguments.directory );
+		// Add directory given (if any)
+		addDirectories( arguments.directories );
+
+		return this;
+	}
+
+	/**
+	* Constructor
+	* @directory A directory to test which can be a simple mapping path or a struct with the following options: [ mapping = the path to the directory using dot notation (myapp.testing.specs), recurse = boolean, filter = closure that receives the path of the CFC found, it must return true to process or false to continue process ]
+	*/
+	any function addDirectory(required any directory, boolean recurse = true) {
 		// inflate directory?
-		if( isSimpleValue( arguments.directory ) ){ arguments.directory = { mapping=arguments.directory, recurse=true }; }
+		if( isSimpleValue( arguments.directory ) ) { arguments.directory = { mapping=arguments.directory, recurse=arguments.recurse }; }
 		// directory passed?
 		if( !structIsEmpty( arguments.directory ) ){
-			arguments.bundles = getSpecPaths( arguments.directory );
+			variables.bundles = ArrayMerge(variables.bundles, getSpecPaths( arguments.directory ));
 		}
+		return this;
+	}
 
-		// inflate labels
-		inflateLabels( arguments.labels );
-		// inflate bundles to array
-		inflateBundles( arguments.bundles );
+	/**
+	* Constructor
+	* @directories A set of directories to test which can be a list of simple mapping paths or an array of structs with the following options: [ mapping = the path to the directory using dot notation (myapp.testing.specs), recurse = boolean, filter = closure that receives the path of the CFC found, it must return true to process or false to continue process ]
+	*/
+	any function addDirectories(required any directories, boolean recurse = true) {
+		// inflate directory?
+		if( isSimpleValue( arguments.directories ) ) { arguments.directories = ListToArray(arguments.directories); }
+		// directory passed?
+		if( !arrayIsEmpty( arguments.directories ) ){
```

Comment:
You can actually eliminate this check, the for loop will just skip if the target is empty